crates/cli/src/lib.rs

```diff
@@ -568,3 +598,34 @@ fn build_ext(
 
     bail!("Failed to retrieve extension path from artifact")
 }
+
+// Write given string to a given filepath.
+//
+// - Write to a temp file first.
+// - Copy the temp file to the given filepath.
+// - If it fails, move tempfile using `sudo mv`.
+//
+// TODO: Try with sudo when error is permission related.
+fn write_to_file(content: String, filepath: &PathBuf) -> anyhow::Result<()> {
+    // write to a temp file
+    let tempf = std::env::temp_dir().join("__tmp_cargo_php");
+    std::fs::write(&tempf, content)?;
+
+    // Now move. `rename` will overwrite existing file.
+    if std::fs::rename(&tempf, filepath).is_err() {
+        #[cfg(unix)]
+        {
+            // if not successful, try with sudo on unix.
+            let _ = std::process::Command::new("sudo")
+                .arg("mv")
+                .arg(&tempf)
+                .arg(filepath)
+                .status()?;
+        }
+
+        #[cfg(not(unix))]
+        anyhow::bail!("failed to write to {filepath:?}");
+    }
```

Comment:
Technically this could override changes made between the time the tmp file is created and its copied.

But I think that is fine for now.

Reply:
I tried adding a write lock, but then I didn't know how to do it properly on Windows without pulling-in `windows-rs` crate.

There is https://crates.io/crates/tempfile that may be a cross-platform solution.